The paper is motivated by the problem of steady states identification as well as reachability problem in biological regulatory systems that describes genes and proteins interactions. Indeed the regulatory phenomena play a crucial role in biological systems, they need to be studied accurately. That’s why the study of Biological Regulatory Networks (BRNs) has been the subject of numerous researches. BRNs consist in sets of either positive or negative mutual effects between the components. With the purpose of analyzing these systems, they are often modeled as graphs which makes it possible to determine the possible evolutions of all the interacting components of the system. Thus, in order to address the formal checking of dynamical properties within very large BRNs, new formalisms and conversely new techniques have been proposed during the last decade. For example, Boolean networks \cite{stuart1993origins} \cite{kauffman1969metabolic} have been used due to their simplicity and ability to handle noisy data but lose data information by having a binary representation of the genes. Besides, artificial neural networks omit using a hidden layer so that they can be interpreted, losing the ability to model higher order correlations in the data. We can cite some works that used these models and which have been developed to verify the dynamic properties of BRNs modeled : GINsim (Gene Interaction Network simulation) \cite{chaouiya2012logical, gonzalez2006ginsim} is a computer tool for the modeling and simulation of genetic regulatory networks. It allows the user to simulate a model and/or analyse its qualitative dynamical behavior. The inconvenient of this tool is the difficulty of analyzing the big networks. Indeed, in order to verifying a dynamical property it is necessary to compute the boolean network from the corresponding Thomas network. We can cite also libDDD (Library of Data Decision Diagrams) a library for symbolic model-checking of CTL \& LTL properties. It can thus especially be used to check reachability properties. But it cannot output the execution path solving the reachability.The computing tests show that \textsc{LibDDD} and \textsc{GINsim} take a long time to respond, and gets out of memory for the big examples (more than 40 components).  %à revoir
%Several other promising modeling and analysing techniques have been used, including Boolean networks, Petri nets, Bayesian networks, we can cite {ref des autres travaux sur l’atteignabilité} \ref{paoletti2014analyzing}

It is also common to model biological networks with a set of coupled ordinary differential equations (ODEs) \cite{chu2009models}, describing the kinetic reactions. But the temporal evolution of the systems modeled by ODEs is computed by a complex derivation approach. This complexity of computing the evolution so that the verification of the dynamic properties causes also a long time to respond. 

New experimental results \cite{elowitz2002stochastic} \cite{blake2003noise} demonstrated that gene expression is a stochastic process. Thus, many biologists and bioinformaticians are now using the stochastic formalism \cite{arkin1998stochastic}. This formalism permits to represent each gene expression \cite{raser2005noise} and small synthetic genetic networks, \cite{elowitz2000synthetic} \cite{gardner2000construction}. Such that our new formalism named Process Hitting (PH) \cite{PMR10-TCSB}. It models concurrent systems having components with a few qualitative levels. Furthermore, PH is suitable, according to the precision of the available information, to model BRNs with different levels of abstraction by capturing the most general dynamics.
Thus, in order to address the formal checking of dynamical properties within very large BRNs, we recently introduced a new formalism, named the "Process Hitting'' (PH) \cite{PMR10-TCSB}, to model concurrent systems having components with a few qualitative levels. A PH describes, in an atomic manner, the possible evolutions of a "process'' (representing one component at one level) triggered by the hit of other "processes’’ in the system. Comparing with other models of BRNs, this particular structure of PH makes the formal analysis of BRNs with hundreds of components easier to be tractable. This was proved by a first work on the PH in \cite{PMR12-MSCS} which analysis big networks and gives a response through a short time. But this developed technic was based on abstract methods computing approximations of the dynamics that can be inconclusive in some cases. Moreover, in the case of a positive answer, it currently does not return the execution of the path achieving the desired reachability, but only outputs its conclusion.

Our goal in this paper is to develop exhaustive methods that analyze Biological Regulatory Network modeled in Process Hitting. We will prove that these methods are exhaustive and fast. In addition it returns the execution path for the reachability problem and all the stable states of a model.

The paper is organised as follows % à remplir
We conclude the paper with a short discussion. 